tools/debug/debugsymbolviewer.py

```diff
@@ -317,6 +324,19 @@ def print_running_work_function(log_file):
                             node = int(wf[:-2], 16)
                             print_symbol(0x00000000, node, 0)
                             print_symbol(0x00000000, node, 1)
+
+# Function to Parse the input log file (which contains interrupt handler data) and to print its owner
+def print_interrupt_handler_data(log_file):
+	# Parse the contents based on tokens in log file.
+	with open(log_file) as searchfile:
+		for line in searchfile:
+			if 'IRQ handler' in line:
+                            word = line.split(' ')
+                            node = int(word[-1], 16)
+                            print_symbol(0x00000000, node, 0)
+                            print_symbol(0x00000000, node, 1)
```

Comment:
This might not work if debug irq config is enabled.

```
up_dumpstate: IRQ handler: 0e00cb29
up_dumpstate: IRQ handler: (xcpt_t)up_timerisr
```

Reply:
I verified it, there is no such issue, it is working fine ..